alb.tf

```diff
@@ -1,83 +1,79 @@
-resource "aws_alb_target_group" "public" {
-  deregistration_delay = 5
-  name                 = "${var.service_name}-public"
-  port                 = var.container_port
-  protocol             = "HTTP"
-  tags                 = local.default_tags
-  target_type          = "ip"
-  vpc_id               = data.aws_vpc.selected.id
+data "aws_lb" "public" {
+  count = var.alb_attach_public_target_group ? 1 : 0
+  name  = "public"
+}
 
-  health_check {
-    healthy_threshold   = 2
-    interval            = 30
-    path                = var.health_check_endpoint
-    port                = var.container_port
-    protocol            = "HTTP"
-    unhealthy_threshold = 10
-  }
+data "aws_lb_listener" "public" {
+  count             = var.alb_attach_public_target_group ? 1 : 0
+  load_balancer_arn = data.aws_lb.public[count.index].arn
+  port              = 443
 }
 
-resource "aws_alb_target_group" "private" {
+data "aws_lb_listener" "public_80" {
+  count             = var.alb_attach_public_target_group ? 1 : 0
+  load_balancer_arn = data.aws_lb.public[count.index].arn
+  port              = 80
+}
+
+resource "aws_alb_target_group" "public" {
+  count = var.alb_attach_public_target_group ? 1 : 0
+
   deregistration_delay = 5
-  name                 = "${var.service_name}-private"
+  name                 = "${var.service_name}-public"
```

Comment:
c&p error.

Reply:
this should be fine: I moved the `aws_alb_target_group.private` a bit lower - guess the diff view doesn't like it